scripts/start_validator.py

```diff
@@ -0,0 +1,199 @@
+    try:
+        subprocess.run(
+            split("git pull --rebase --autostash"), check=True, cwd=CURRENT_WORKING_DIR
+        )
+    except subprocess.CalledProcessError as exc:
+        log.error("Failed to pull, reverting: %s", exc)
+        subprocess.run(split("git rebase --abort"), check=True, cwd=CURRENT_WORKING_DIR)
+
+
+def upgrade_packages() -> None:
+    """
+    Upgrade python packages by running `pip install --upgrade -r requirements.txt`.
+    Notice: this won't work if some package in `requirements.txt` is downgraded.
+    Ignored as this is unlikely to happen.
+    """
+    log.info("Upgrading packages")
+    try:
+        subprocess.run(
+            split(f"{sys.executable} -m pip install --upgrade -r requirements.txt"),
+            check=True,
+            cwd=CURRENT_WORKING_DIR,
+        )
+    except subprocess.CalledProcessError as exc:
+        log.error("Failed to upgrade packages, proceeding anyway. %s", exc)
+
+
+def main(pm2_name: str, args_namespace: Namespace) -> None:
+    """
+    Run the validator process and automatically update it when a new version is released.
+    This will check for updates every `UPDATES_CHECK_TIME` and update the validator
+    if a new version is available. Update is performed as simple `git pull --rebase`.
+    """
+
+    args_list = []
+    for key, value in vars(args_namespace).items():
+        if value != '' and value is not None:
+            args_list.append(f"--{key}")
+            if not isinstance(value, bool):
+                args_list.append(str(value))
+
+    validator = start_validator_process(pm2_name, args_list)
+    current_version = latest_version = get_version()
+    log.info("Current version: %s", current_version)
+
+    try:
+        while True:
+            pull_latest_version()
+            latest_version = get_version()
+            log.info("Latest version: %s", latest_version)
+
+            if latest_version != current_version:
+                log.info(
+                    "Upgraded to latest version: %s -> %s",
+                    current_version,
+                    latest_version,
+                )
+                upgrade_packages()
+
+                stop_validator_process(validator)
+                validator = start_validator_process(pm2_name, args_list)
+                current_version = latest_version
+
+            time.sleep(UPDATES_CHECK_TIME.total_seconds())
+
+    finally:
+        stop_validator_process(validator)
+
+
+if __name__ == "__main__":
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s %(levelname)s %(message)s",
+        handlers=[logging.StreamHandler(sys.stdout)],
+    )
+
+    parser = argparse.ArgumentParser(
+        description="Automatically update and restart the validator process when a new version is released.",
+        epilog="Example usage: python start_validator.py --pm2_name 'net9vali' --wallet_name 'wallet1' --wallet_hotkey 'key123'",
+    )
+
+    parser.add_argument(
```

Comment:
remove args, and put in documentation that  you need to edit pm2 file.

Also move pm2 file to config directory